docs/proof_object.md

```diff
@@ -275,3 +275,91 @@ proof checker extremely simple.
 * Check that `P11` is syntactically equal to `Q1`
 * Check that `P12` is syntactically equal to `P21`
 * Check that `P22` is syntactically equal to `Q2`
+
+# Mixed-level Proof Object Syntax
+
+Language definitions are expected to use mostly
+object-level or mixed-level formulas, which is why
+we have defined the Kore syntax to allow these formulas,
+rather than simply requiring everything to be encoded
+in meta-level formulas.
+For the same reason, we wish to define a syntax of proofs
+that allows writing mixed-level formulas, and that allows
+object-level reasoning to be written in the
+form `by <ProofRule>` rather than by explicitly
+constructing meta-level proofs of terms involving `#provable`
+using the hypotheses described in section 7.7 of
+The Semantics of K.
+
+To support this we allow as patterns any Kore pattern,
+according to the syntax of `<pattern>` in 9.1.4.
+```
+  <ProofObject>  ::= <AssumedClaim>* <DerivedClaim>*
+
+  <AssumedClaim> ::= <Id> : <Pattern>
+
+  <DerivedClaim> ::= <Id> : <Pattern> by <ProofRule>
+
+  <Id> ::= /* Quite flexible syntax. */
+```
+
+Proofs will be defined so that a valid mixed-level proof can
+be transformed into a valid simple proof that assumes or
+proves (resp.) the lifting of each assumed or derived claim
+of the mixed-level proof.
+
+#### Note
+We may wish to disambiguate the object-level proof rule
+
+A pattern is considered equivalent to its lifting.
+Following the convention of lifting we will allow some
+proof rule names to be re-used between object-level and meta-level
+forms, prefixing a '#' to unambiguously refer to the meta-level form.
+
+All the proof rules above should be available at the object level as well,
+but now the actual reasoning will be in the form of properly
+instantiating and applying a corresponding meta-level hypothesis.
+
+The "object-level" form of the proof rules are equivalent to proving
+`#provable` of the lifting of the formula.
+To exactly resemble the meta-level proof rules, the `#wellFormed`
+hypotheses are automatically discharged.

```

Comment:
Why are they automatically discharged?

Reply:
As it says, "To exactly resemble the meta-level proof rules".
A rule like the matching logic "singvar" doesn't have an explicit hypothesis about the formulas being well-formed.